PollBuddy-Server/backend/modules/utils.js

```diff
@@ -45,6 +46,20 @@ function isLoggedIn(req) {
   return req.session.userData && req.session.userData.userID;
 }
 
+/**
+ * Convenience function to get the currently logged in user
+ * Dumps result into passed callback function
+ * @returns {void} 
+ * @name getCurrentUser
+ * @param {req} req request object
+ * @param {callback} callback handler for (err,result) returned by database query
+ */
+function getCurrentUser(req,callback) {
+  mongoConnection.getDB().collection("users").findOne({ _id: bson.ObjectId(req.session["UserName"]) }, { projection: { Password: false } }, (err, result) => {
```

Comment:
Looking good! There's just one thing, req.session["UserName"] has been changed to req.session.userData.userID.